Add BoostNoteNext v0.9.0

J3n50m4t opened the pull request:
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download boost-note-next` is error-free.
- [x] `brew cask style --fix boost-note-next` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask boost-note-next` worked successfully.
- [x] `brew cask install boost-note-next` worked successfully.
- [x] `brew cask uninstall boost-note-next` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

NOTE: 
I did not name it accorting to the token reference, because it would lead to `boost-note`. 
Actually there already is `boostnote`, which is the old version. To my understanding `boostnote`should be named boost-note and the added one is named BoostNode.Nextm which doesn't seem to have an Naming convention? Happy to get enlightened.

miccal commented:
> NOTE:
> I did not name it accorting to the token reference, because it would lead to `boost-note`.
> Actually there already is `boostnote`, which is the old version. To my understanding `boostnote`should be named boost-note and the added one is named BoostNode.Nextm which doesn't seem to have an Naming convention? Happy to get enlightened.

The [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md) uses the exact name of the application bundle as it appears on disk, so [`boostnote`](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/boostnote.rb) is correctly named as its application bundle is called `Boostnote.app`, and therefore this one should indeed be called `boost-note` because its application bundle is called `Boost Note.app`.

miccal commented on Casks/boost-note-next.rb:
```diff
@@ -0,0 +1,23 @@
+  url "https://github.com/BoostIO/BoostNote.next/releases/download/v#{version}/boost-note-mac.dmg"
+  appcast "https://github.com/BoostIO/BoostNote.next/releases.atom"
+  name "Boostnote.Next"
+  desc "Intuitive and stylish markdown note app for the developers"
```
```suggestion
  desc "Markdown note editor for developers"
```

miccal commented on Casks/boost-note-next.rb:
```diff
@@ -0,0 +1,23 @@
+
+  app "Boost Note.app"
+
+  uninstall signal: [
+    ["TERM", "com.boostio.boostnote"],
+  ]
```
```suggestion
  uninstall signal: ["TERM", "com.boostio.boostnote"]
```

J3n50m4t commented:
I've added your suggestions and renamed the boost-note-next.rb to boost-note.rb.
I must have been confused by to many Boost Note.app Files and must have chosen the wrong one for token-reference. Thanks for enlightening me.

miccal commented:
Thank you for your contribution to `homebrew-cask` @J3n50m4t.
